### PR TITLE
Added autoFlipDuration parameter

### PR DESCRIPTION
**Pull request by aydinfatih**
# Description

Added one-time autoFlip feature without a FlipCardController. For example, a one-time flip can be played for the logo or the cards that need to be loaded once.

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

I tested it on my own project. The following parameter should be added to the FlipCard widget.

` autoFlipDuration: Duration(milliseconds: 1000),`

- [x] Self Code Review
- [x] Tested on example project

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Changed the example project to contain the new feature (only if applies)

### CODE REVIEW

**Comment by BrunoJurkovic** [comment]
Thanks for your enhancement and for your time, and my apology for the delayed response. This seems like a nice little feature to add. @aydinfatih Could you please write some documentation for this? Thank you :)

File: lib/flip_card.dart [hunk 1]
```diff
@@ -155,6 +157,10 @@ class FlipCardState extends State<FlipCard>
     ).animate(controller!);
 
     widget.controller?.state = this;
+
+    if (widget.autoFlipDuration != null) {
+      Future.delayed(widget.autoFlipDuration!, () => toggleCard());
```
Comment (ciriousjoker) [inline lib/flip_card.dart]: You can also just pass `toggleCard` directly as the second parameter (without creating an anonymous function to call it yourself)

File: lib/flip_card.dart [hunk 2]
```diff
@@ -60,6 +60,7 @@ class FlipCard extends StatefulWidget {
   final FlipCardController? controller;
   final Fill fill;
   final CardSide side;
+  final Duration? autoFlipDuration;
```
Comment (ciriousjoker) [inline lib/flip_card.dart:65]: Please documentation for this feature, ideally as a `/// comment` above the autoFlipDuration, but also in the readme

**Review by ciriousjoker — CHANGES_REQUESTED** [review]

**Comment by ciriousjoker** [comment]
@aydinfatih I'd like to close this soon. Can you add the documentation for it please?

File: lib/flip_card.dart (hunk 2 above)
Comment (aydinfatih) [reply lib/flip_card.dart:65]: Added to documentation.

File: lib/flip_card.dart (hunk 1 above)
Comment (aydinfatih) [reply lib/flip_card.dart]: I updated it.

**Comment by aydinfatih** [comment]
@ciriousjoker Sorry for late. I made the desired changes.

File: README.md [hunk 3]
```diff
@@ -84,3 +84,19 @@ void doStuff() {
   _controller.toggleCard();
 }
 ```
+
+You can auto-flip the widget after a certain delay without any external triggering.
+```dart
+FlipCard(
+  fill: Fill.fillBack, // Fill the back side of the card to make in the same size as the front.
```
Comment (ciriousjoker) [inline README.md:91]: I'd probably combine all of these into

```
(
  ...,
  autoFlipDuration ...
)
```

File: lib/flip_card.dart [hunk 4]
```diff
@@ -61,6 +61,9 @@ class FlipCard extends StatefulWidget {
   final Fill fill;
   final CardSide side;
 
+  /// If the value is set, the flip effect will work automatically after the specified duration.
+  final Duration? autoFlipDuration;
```
Comment (ciriousjoker) [inline lib/flip_card.dart:65]: It looks like formatting is broken? Reformat or ignore this

**Review by ciriousjoker — APPROVED** [review]
Lgtm, just some minor improvements

**Comment by ciriousjoker** [comment]
@aydinfatih I'll probably merge this at the end of next week. If you make any changes until then I'll review them, otherwise consider it merged.

File: lib/flip_card.dart (hunk 4 above)
Comment (aydinfatih) [reply lib/flip_card.dart:65]: What is the problem here? I do not fully understand.

Comment (ciriousjoker) [reply lib/flip_card.dart:65]: Ah nevermind. I was looking at this on mobile and it looked like the formatting was weird, but it just looked like that for me.

This is completely fine!

**Comment by ciriousjoker** [comment]
@BrunoJurkovic Can you please publish a new version to pub.dev?

**Pull request merged**